### PR TITLE
Elasticsearch addition to zeeguu

### PR DESCRIPTION
**Pull request by Mgp1502**
- Added an elastic_recommender that can be used instead of the mixed_recommender.
- Added tool for index database to elastic to allow for searches (mysql_to_elastic).
- Added so the article retriever also saves new articles in elasticsearch.
- Updated readme with elastic search part.

### CODE REVIEW

File: zeeguu_core/content_retriever/article_downloader.py [hunk 1]
```diff
@@ -147,14 +149,39 @@ def download_from_feed(feed: RSSFeed, session, limit=1000):
...
+                        # as ElasticSearch isn't persistant data
+                        try:
+                            if save_in_elastic:
+                                es = Elasticsearch([settings["ip"]])
+                                doc = {
+                                    'title': new_article.title,
+                                    'author': new_article.authors,
+                                    'content': new_article.content,
+                                    'summary': new_article.summary,
+                                    'word_count': new_article.word_count,
```
Comment (mircealungu) [inline zeeguu_core/content_retriever/article_downloader.py]: please remove duplication if possible.

File: zeeguu_core/content_retriever/article_downloader.py [hunk 2]
```diff
@@ -176,6 +203,13 @@ def download_from_feed(feed: RSSFeed, session, limit=1000):
     log(f'  Already in DB: {skipped_already_in_db}')
 
 
+def topics_tostring(topics):
+    res = ""
+    for t in topics:
+        res += " " + t.title
```
Comment (mircealungu) [inline zeeguu_core/content_retriever/article_downloader.py]: ' '.join([each.title for each in topics]) or ''

File: README.md [hunk 3]
```diff
@@ -38,7 +37,22 @@ when installing mysqlclient try to:
     
 this is cf. https://stackoverflow.com/a/51701291/1200070
 
+# Setting up ElasticSearch (Mac / Linux)
+Follow instructions on: 
+  https://www.elastic.co/guide/en/elasticsearch/reference/current/targz.html
```
Comment (mircealungu) [inline README.md:42]: Which brings me to: which version of ElasticSearch did you test with? 
I see that there's  both 6 and 7 still around.

File: README.md [hunk 4]
```diff
@@ -38,7 +37,22 @@ when installing mysqlclient try to:
...
+
+Install on localhost for test (127.0.0.1:9200)
+
+When installing, we recommend at least 4 GB dedicated to ElasticSearch. This can easily support querying of 30+ concurrent users. If it is a single node cluster (one server), we recommend the node to both be able to ingest, hold data and be master. The two latter needs to be enabled. 
+
+To export data from MySQL to ElasticSearch run  Zeeguu_Core/Tools/mysql_to_elastic.py. Please notice that the name of the index is placed in elastic_settings.py located in the main folder of Zeeguu Core.
```
Comment (mircealungu) [inline README.md:48]: i don't see this `elastic_settings.py` in the PR. 
did you push it?

Comment (mircealungu) [inline README.md:48]: fix case pls: zeeguu_core/tools/mysql_to_elastic.py

File: elastic/converting_from_mysql.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+from zeeguu_core.model import Topic
```
Comment (mircealungu) [inline elastic/converting_from_mysql.py:1]: i like that you tried to separate this from zeeguu_core but running setup.py install on this thing will install it as the elastic package and that's too risky -- there might be other packages with the same name.

i propose we move it under `zeeguu_core.elastic` instead. 

agree?

Comment (Mgp1502) [reply elastic/converting_from_mysql.py]: Makes sense.
Changed it to this in one of the new commits.

File: README.md (hunk 4 above)
Comment (Mgp1502) [reply README.md]: This should now be located under zeeguu_core.elastic.settings.py

File: README.md (hunk 3 above)
Comment (Mgp1502) [reply README.md:42]: We use version 7.6

File: zeeguu_core/content_recommender/elastic_recommender.py [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+    """
+    article = fetch_article_by_ID(article_id)
+
+    query_body = build_more_like_this_query(count, article.content, article.language)
+
+    es = Elasticsearch(ES_CONN_STRING)
+    res = es.search(index=ES_ZINDEX, body=query_body)  # execute search
+    hit_list = res['hits'].get('hits')
+
+    # TODO need to make sure either that the searched on article is always a part of the list \
```
Comment (mircealungu) [inline zeeguu_core/content_recommender/elastic_recommender.py:33]: Should this TODO still be here?

Comment (Mgp1502) [reply zeeguu_core/content_recommender/elastic_recommender.py:33]: It is more meant as a reminder if/when more_like_this get implemented on the frontend to look at this case and decide what to do. 
Could also make an issue on it instead if you would rather have that.

Comment (mircealungu) [reply zeeguu_core/content_recommender/elastic_recommender.py:33]: But I think the original article should never be in the set of "more like this". Or?

Comment (mircealungu) [reply zeeguu_core/content_recommender/elastic_recommender.py]: also, make sure to pull before making any further changes to the code. i've refactored it a bit.

**Pull request merged**